src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.cs

```diff
@@ -69,21 +50,28 @@ protected virtual SyntaxNode GetNodeToAnalyze(SyntaxNode root, TextSpan span)
             return actions;
         }
 
+        private bool ShouldAnalyze(SyntaxNode root, TextSpan span) =>
+            GetNodeToAnalyze(root, span) is TTypeDeclarationSyntax;
+
         private ImmutableArray<CodeAction> CreateActions(State state, CancellationToken cancellationToken)
         {
             var actions = new List<CodeAction>();
             var manyTypes = MultipleTopLevelTypeDeclarationInSourceDocument(state.SemanticDocument.Root);
 
+            // (1) Add Move type to new file code action:
+            // case 1: There are multiple type declarations in current document. offer, move to new file.
+            // case 2: This is a nested type, offer to move to new file.
+            // case 3: If there is a single type decl in current file, *do not* offer move to new file.
             if (manyTypes || IsNestedType(state.TypeNode))
             {
-                // If there are multiple type declarations in current document. offer, move to new file.
-                // Or if this is a nested type, offer to move to new file.
                 actions.Add(GetCodeAction(state, operationKind: OperationKind.MoveType));
             }
-            else
+
+            // (2) Add rename file and rename type code actions:
+            // case 1: Multiple type decls in file and none match file name.
+            // case 2: Single type declaration in file and names don't match.
```

Comment:
Case 1 and 2 are the same "No type declaration in the file matches the file name."